### ISSUE TITLE
Get block stateId from properties (1.13+)

### ISSUE THREAD

**Issue by TheDudeFromCI** [PrismarineJS/prismarine-block#28]
In cases like working with staircases or half slabs, there are many times when it extremely useful to think of blocks in the form of their type and state properties. For example, if I wanted to get a top wooden oak half slab that is waterlogged, it's not very easy to determine what the state id I need is.

You can kind of work with this by iterating between the minStateId and maxStateId and checking against all the properties, but this is not ideal.

I recommend something along the lines of `Block.fromProperties(typeId, properties, biomeId)`

**Comment by Karang** [comment]
Yeah this function is implemented in provider-anvil and schematic, but it would be nice to have it here only once:
* https://github.com/PrismarineJS/prismarine-provider-anvil/blob/master/src/1.13/chunk.js#L80
* https://github.com/PrismarineJS/prismarine-schematic/blob/master/lib/states.js#L32
Notice that we could only do that for 1.13+ we don't have a source for the state data before that (it was hardcoded).

**Comment by TheDudeFromCI** [comment]
> Yeah this function is implemented in provider-anvil and schematic, but it would be nice to have it here only once:
> 
>     * https://github.com/PrismarineJS/prismarine-provider-anvil/blob/master/src/1.13/chunk.js#L80
> 
>     * https://github.com/PrismarineJS/prismarine-schematic/blob/master/lib/states.js#L32
> 
> 
> Notice that we could only do that for 1.13+ we don't have a source for the state data before that (it was hardcoded).

Oh, so it's basically just copying and paste, huh? That makes it pretty easy.
Yeah, I know it only works on 1.13+, (hints the title, lol) so we can simply throw an error on earlier versions.
